File: DataBinding/PageAndRDL/XmlDataSource/C#/XmlDataSource/DataLayer.cs
using System.Xml;
using System.Xml.XPath;

namespace GrapeCity.ActiveReports.Samples.XmlDataSource
{
	// Provides the data used in the sample.
	internal sealed class DataLayer
	{
		public XmlReader CreateReader()
		{
			var txtReader = new XmlTextReader(@"..\..\MyXmlDB.xml");
			return txtReader;
		}

		public IXPathNavigable CreateDocument()
		{
			var doc = new XPathDocument(@"..\..\MyXmlDB.xml");
			return doc;
		}
	}
}

File: DataBinding/PageAndRDL/XmlDataSource/VB.NET/XmlDataSource/DataLayer.vb
Imports System.Xml
Imports System.Xml.XPath

' Provides the data used in the sample.
Friend NotInheritable Class DataLayer

	Public Function CreateReader() As XmlReader
		Dim txtReader As New XmlTextReader("..\..\MyXmlDB.xml")
		Return txtReader
	End Function

	Public Function CreateDocument() As IXPathNavigable
		Dim doc As New XPathDocument("..\..\MyXmlDB.xml")
		Return doc
	End Function

End Class
